### PR TITLE
Leg implants

### PR DESCRIPTION
**Pull request by Moltijoe**
Adds leg implants
Lets you do organ manipulation on the legs

- [x] Water slip immunity
- [x] Complete slip immunity (not currently obtainable, can't find a suitable place)
- [x] Maglock implant
- [x] Add research and printing costs

default leg implant icon is just the arm implant rotated 90 degrees, someone with more artistic skill should probably make a proper one eventually
image

Ones other people could make (not me, i just want the maglock in so i can use it for my preternis rework)
Rocket jump implant
Clown shoes squeak implant
other walk sound implants?
Table climb speed implant?
Movement speed implant?

:cl:  
rscadd: Added leg implants
/:cl:

### CODE REVIEW

**Comment by Moggram** [comment]
Rocket jump implants when?

**Comment by Moltijoe** [comment]
Done with this PR, will probably make more implants down the road though

**Comment by VaelophisNyx** [comment]
total slip immunity would make for a good traitor implant briefcase; ~8tc for total slip immunity?

**Comment by Moltijoe** [comment]
> total slip immunity would make for a good traitor implant briefcase; ~8tc for total slip immunity?

I considered it, but couldn't think of any good number for balancing since it seems like it could either be super op, or far less useful than it seems

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 1]
```diff
@@ -38,6 +38,8 @@
 		var/obj/item/clothing/S = shoes
 		if((S.clothing_flags & NOSLIP))
 			return 0
+	if(HAS_TRAIT(src, TRAIT_MAGBOOT))
+		return 0
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/human/human_movement.dm]: TRUE/FALSE

File: code/modules/surgery/organs/augments_legs.dm [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		transform = matrix(-1, 0, 0, 0, 1, 0)
+
+/obj/item/organ/cyberimp/leg/examine(mob/user)
+	. = ..()
+	. += span_info("[src] is assembled in the [zone == BODY_ZONE_R_LEG ? "right" : "left"] leg configuration. You can use a screwdriver to reassemble it.")
+
+/obj/item/organ/cyberimp/leg/screwdriver_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(.)
+		return TRUE
```
Comment (Bibby0110) [inline code/modules/surgery/organs/augments_legs.dm]:
```suggestion
		return .
```

File: code/modules/surgery/organs/augments_legs.dm [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	I.play_tool_sound(src)
+	if(zone == BODY_ZONE_R_LEG)
+		zone = BODY_ZONE_L_LEG
+	else
+		zone = BODY_ZONE_R_LEG
+	SetSlotFromZone()
+	to_chat(user, span_notice("You modify [src] to be installed on the [zone == BODY_ZONE_R_LEG ? "right" : "left"] leg."))
+	update_icon()
+
+/obj/item/organ/cyberimp/leg/Remove(mob/living/carbon/M, special = 0)
```
Comment (Bibby0110) [inline code/modules/surgery/organs/augments_legs.dm]: unneeded

File: code/modules/surgery/organs/augments_legs.dm [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	var/active = FALSE
+	actions_types = list(/datum/action/item_action/organ_action/toggle)
+
+/obj/item/organ/cyberimp/leg/maglock/l
+	zone = BODY_ZONE_L_LEG
+
+/obj/item/organ/cyberimp/leg/maglock/Destroy()
+	if(active)
+		ui_action_click()
+	..()
```
Comment (Bibby0110) [inline code/modules/surgery/organs/augments_legs.dm]:
```suggestion
	. = ..()
```

File: code/modules/surgery/organs/augments_legs.dm [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+/obj/item/organ/cyberimp/leg/noslip/Insert()
+	. = ..()
+	ADD_TRAIT(owner, TRAIT_NOSLIPALL, "Noslip_implant")
+
+/obj/item/organ/cyberimp/leg/noslip/Remove(mob/living/carbon/M, special = FALSE)
+	REMOVE_TRAIT(owner, TRAIT_NOSLIPALL, "Noslip_implant")
+	. = ..()
+
+// /obj/item/organ/cyberimp/leg/clown
```
Comment (Bibby0110) [inline code/modules/surgery/organs/augments_legs.dm]: remove this if youre keeping it commented

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
e

**Review by KillerOrcaCora — APPROVED** [review]

**Comment by JamieD1** [comment]
Kinda feels a bit op

**Comment by Skrem7** [comment]
> Kinda feels a bit op

I think it would be more fitting if you had to double-up on the same type, one for each leg, otherwise, it doesn't work. Not seeing anything like that at the moment. Magboot implant is 100% fine, it's the no-slips that are a bit more questionable, but there's already plenty of researched implants and organs to make things like thermal goggles or the CNS autosurgeon redundant. Certainly on the scarier side with the no-slips, but it could probably be surveyed for a while, make sure Sec isn't using it to too much of an advantage.

**Comment by Moltijoe** [comment]
> Kinda feels a bit op

Aside from what skrem said, i cant really think of anything to make them less op. The requiring doubles thing will definitely take some time to get working.

Having maglocking without needing the boots seems rather niche in use. I will agree i did think the nowaterslip was a bit more powerful so i locked it behind combat implants just like antidrop is, which it seemed a side grade to.

**Comment by Skrem7** [comment]
> The requiring doubles thing will definitely take some time to get working.

Not too much of an issue in my opinion. I can do it later if I have to. Just consider how awkward it might be for preterni if they get one magboot implant per leg and get two action buttons (though I'm pretty sure that's not how it works).

> Having maglocking without needing the boots seems rather niche in use. I will agree i did think the nowaterslip was a bit more powerful so i locked it behind combat implants just like antidrop is, which it seemed a side grade to. 

Completely agree here. It feels good to get upgraded. Augmentation has its place in the game and EMPs are a solid counter to it. This just expands the list of options.

**Comment by Moltijoe** [comment]
> 
> Not too much of an issue in my opinion. I can do it later if I have to. Just consider how awkward it might be for preterni if they get one magboot implant per leg and get two action buttons (though I'm pretty sure that's not how it works).
> 
 
I can always just update the preternis code to be nicer instead if these implants dont get in. Also id probably just have it grant the ability once there is two of the same implant in both legs, and remove it when even one gets removed

**Comment by Moltijoe** [comment]
Might re-open this in the future, definitely needs more work before it's worth putting in

**Pull request closed without merging**